### DISCUSSION TITLE
malamute melt pool 3d example

### DISCUSSION THREAD

**Discussion by mcp2001** [idaholab/moose#24027] [Q&A Getting Started] [answered]
Hello,

I am in the process of setting up moose and malamute with the intent to run the melt_pool_3d example from malamute. When I run this, it runs for a single time step, and then stops until I stop the simulation. I am not sure what the problem is and any advice is appreciated.

**Comment by GiudGiud** [comment]
@cticenhour

**Comment by jiangwen84** [comment, marked as answer]
@mcp2001 did you mean the simulation terminated? What error did you see? 

3D example is really expensive to run, and it requires significant memory because of LU solver. I suspect that your memory is not sufficient. If that is the case, could you run 2D example? We have been working to improve the code efficiency.

**Reply by mcp2001** [reply]
Hello and thanks for the reply,
The 2D example runs and I have run it for the full 1000 time steps with no problems. On the 3D example, there is not actually an error that happens, just after the Jacobian calculates the simulation stops computing and stays at that point for hours without progressing. 
![3d one time step](https://user-images.githubusercontent.com/113554269/231213852-ce434fcd-22b9-4db8-a366-7149e206e0fb.png)

**Reply by jiangwen84** [reply]
Could you check your memory usage? I suspect it uses swap memory or virtual memory, which makes it extremely slow.

**Reply by lindsayad** [reply]
Could you also run with `Problem/error_on_jacobian_nonzero_reallocation=true`?

**Reply by mcp2001** [reply]
Thank you very much, it was the memory that was the problem. What amount of memory would be recommended for this 3D simulation?
